ttnn/cpp/ttnn/operations/ccl/all_gather/device/kernels/dataflow/worker_interleaved_ring_gather_receive_reader.cpp

```diff
@@ -9,20 +9,25 @@
 
 
 void kernel_main() {
-    constexpr uint32_t num_transfers = get_compile_time_arg_val(0);
-    constexpr uint32_t num_full_chunks = get_compile_time_arg_val(1);
-    constexpr uint32_t page_size = get_compile_time_arg_val(2);
-    constexpr uint32_t num_pages_per_full_chunk = get_compile_time_arg_val(3);
-    constexpr uint32_t rem_num_pages = get_compile_time_arg_val(4);
-    constexpr uint32_t eth_receiver_noc_x = get_compile_time_arg_val(5);
-    constexpr uint32_t eth_receiver_noc_y = get_compile_time_arg_val(6);
-    constexpr uint32_t eth_receiver_l1_semaphore_addr = get_compile_time_arg_val(7);
-    volatile uint32_t *receiver_read_sem_addr = reinterpret_cast<volatile uint32_t *>(get_semaphore(get_compile_time_arg_val(8)));
-    constexpr uint32_t half_cb_n_pages = get_compile_time_arg_val(9);
-    constexpr uint32_t num_buffers_per_channel = get_compile_time_arg_val(10);
-    static_assert (half_cb_n_pages > rem_num_pages, "half_cb_n_pages must be greater than 0");
-
-    const uint32_t eth_receiver_l1_base_addr = get_arg_val<uint32_t>(0);
+    constexpr uint32_t page_size = get_compile_time_arg_val(0);
+    volatile uint32_t *receiver_read_sem_addr = reinterpret_cast<volatile uint32_t *>(get_semaphore(get_compile_time_arg_val(1)));
+    constexpr uint32_t half_cb_n_pages = get_compile_time_arg_val(2);
+    constexpr uint32_t num_buffers_per_channel = get_compile_time_arg_val(3);
+    // static_assert (half_cb_n_pages > rem_num_pages, "half_cb_n_pages must be greater than 0");
+
+    uint32_t arg_idx = 0;
+    const uint32_t eth_receiver_l1_base_addr = get_arg_val<uint32_t>(arg_idx++);
+    const bool receiver_enabled = get_arg_val<uint32_t>(arg_idx++) == 1;
+    if (!receiver_enabled) {
```

Comment:
Why do we need this? Either kernel shouldn't be invoked or if it just happens to have no work, it'll naturally skip the loops below anyways?

Saves the extra code and extra arg from being passed.